test/OpenTelemetry.Instrumentation.AWS.Tests/Implementation/AWSServiceHelperTests.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Amazon;
+using Amazon.DynamoDBv2;
+using OpenTelemetry.Instrumentation.AWS.Implementation;
+using OpenTelemetry.Instrumentation.AWS.Tests.Tools;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.AWS.Tests.Implementation;
+
+public class AWSServiceHelperTests
+{
+    [Fact]
+    public void GetDynamoDbCloudRegion_RegionEndpointIsSet_CloudRegionIsCorrect()
+    {
+        var region = "eu-central-1";
+        var clientConfig = new AmazonDynamoDBConfig
+        {
+            RegionEndpoint = RegionEndpoint.GetBySystemName(region),
+        };
+        var requestContext = new MockRequestContext(clientConfig);
+
+        var cloudRegion = AWSServiceHelper.GetDynamoDbCloudRegion(requestContext);
+
+        Assert.Equal(region, cloudRegion);
+    }
+
+    [Theory]
+    [InlineData("https://dynamodb.eu-central-1.amazonaws.com")]
+    [InlineData("https://a1b2c3.execute-api.eu-central-1.amazonaws.com/default/mydynamodb")]
+    public void GetDynamoDbCloudRegion_ServiceUrlIsSet_CloudRegionIsCorrect(string serviceUrl)
+    {
+        var clientConfig = new AmazonDynamoDBConfig
+        {
+            ServiceURL = serviceUrl,
+        };
+        var requestContext = new MockRequestContext(clientConfig);
+
+        var cloudRegion = AWSServiceHelper.GetDynamoDbCloudRegion(requestContext);
+
+        Assert.Equal("eu-central-1", cloudRegion);
+    }
+
+    [Fact]
+    public void GetDynamoDbCloudRegion_RegionIsMissingInServiceUrl_CloudRegionIsNull()
+    {
+        var clientConfig = new AmazonDynamoDBConfig
+        {
+            ServiceURL = "https://mycompany.com/test",
```

Comment:
I think that you are missing here something like `https://fooo.bar.mycompany.com/test` - region will be `bar`.
Same for `https://for.baz.bar.mycomany.com/aaaa/bbbb`

Reply:
I would prefer to keep this code as it "documents" default returned by `AWSSDKUtils`. The default is the concrete region us-east-1 that surprised me.